### PR TITLE
FUSETOOLS2-1990: Create UI Tests for code navigation to Camel Contexts and Routes

### PR DESCRIPTION
**Pull request by pospisilf**
# Pull Request informations

## Rebase & Merge default requirements

1. Green build for main branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **main** branch build

### CODE REVIEW

**Comment by pospisilf** [comment]
Blocked by https://github.com/camel-tooling/camel-lsp-client-vscode/pull/1418

File: src/ui-test/tests/code.navigation.test.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        await VSBrowser.instance.waitForWorkbench();
+
+        await waitUntilExtensionIsActivated(driver, `${pjson.displayName}`);
+
+        await (await new ActivityBar().getViewControl('Explorer')).openView();
+    });
+
+    describe('XML DSL', function () {
+
+        // All vailable symbols in 'camel-route.xml' with line of occurence.
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:47]: type: `available`

File: src/ui-test/tests/code.navigation.test.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        const XML_av_symobls = [
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:48]: typo: `symbols`

File: src/ui-test/tests/code.navigation.test.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+            });
+
+            it('goto symbols', async function () {
+                await gotoSymbolsUsingOutlineSideBar(XML_av_symobls);
+            });
+        });
+    });
+
+    describe('Java DSL', function () {
+        const JAVA_av_symobls = [
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:90]: typo: `symbols`

File: src/ui-test/tests/code.navigation.test.ts [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+     * TODO
+     * 
+     * @param listOfAvailableSymbols 
+     */
+    async function gotoSymbolsUsingOutlineSideBar(listOfAvailableSymbols: any): Promise<void> {
+        (await new SideBarView().getContent().getSection('resources')).collapse(); // musi bejt
+
+        const section = await new SideBarView().getContent().getSection('Outline') as DefaultTreeSection; // outline vec
+        await section.expand();
+        await section.isExpanded(); // roztahnout
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:199]: czech comments 😄

File: src/ui-test/tests/code.navigation.test.ts [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        describe('Outline', function () {
+            it('all symbol proposals are avaialble', async function () {
+                await allSymbolsAreAvaialbleInOutlineSideBar(JAVA_av_symobls);
+            });
+
+            it('goto symbols', async function () {
+                await allSymbolsAreAvaialbleInOutlineSideBar(JAVA_av_symobls);
+            });
+        });
+
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:128]: redundant empty line

File: src/ui-test/tests/code.navigation.test.ts [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+            await waitUntilEditorIsOpened(driver, CODE_NAVIGATION_JAVA);
+        });
+
+        afterEach(async function () {
+            await closeEditor(CODE_NAVIGATION_JAVA, false);
+        });
+
+        describe('Symbols', function () {
+
+            it('all symbol proposals are avaialble', async function () {
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:110]: typo: `available`

File: src/ui-test/tests/code.navigation.test.ts [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+                await allSymbolsAreAvaialbleInQuickpickCommand(JAVA_av_symobls);
+            });
+
+            it('goto symbols', async function () {
+                await gotoSymbolsUsingQuickpickCommand(JAVA_av_symobls);
+            });
+        });
+
+        describe('Outline', function () {
+            it('all symbol proposals are avaialble', async function () {
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:120]: typo: `available`

File: src/ui-test/tests/code.navigation.test.ts [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+                await allSymbolsAreAvaialbleInQuickpickCommand(XML_av_symobls);
+            });
+
+            it('goto symbols', async function () {
+                await gotoSymbolsUsingQuickpickCommand(XML_av_symobls);
+            });
+        });
+
+        describe('Outline Side Bar navigation', function () {
+            it('all symbol proposals are avaialble', async function () {
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:79]: typo: `available`

File: src/ui-test/tests/code.navigation.test.ts [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+            await waitUntilEditorIsOpened(driver, CAMEL_ROUTE_XML);
+        });
+
+        afterEach(async function () {
+            await closeEditor(CAMEL_ROUTE_XML, false);
+        });
+
+        describe('Quickpick \'Go to Symbol in Editor\' navigation', function () {
+
+            it('all symbol proposals are avaialble', async function () {
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:69]: typo: `available`

File: src/ui-test/tests/code.navigation.test.ts [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { assert } from "chai";
+import { ActivityBar, DefaultTreeSection, InputBox, QuickPickItem, SideBarView, TextEditor, VSBrowser, WebDriver, Workbench } from "vscode-uitests-tooling";
+import { CAMEL_ROUTE_XML, CODE_NAVIGATION_JAVA, RESOURCES, closeEditor, waitUntilEditorIsOpened, waitUntilExtensionIsActivated } from "../utils/testUtils";
+import * as pjson from '../../../package.json';
+import path = require("path");
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:22]: better to use `import * as path from 'path'`

File: src/ui-test/tests/code.navigation.test.ts (hunk 4 above)
Comment (pospisilf) [reply src/ui-test/tests/code.navigation.test.ts:199]: labels: DRAFT 😅

File: src/ui-test/tests/code.navigation.test.ts [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+            });
+        });
+
+        describe('Outline', function () {
+            it('all symbol proposals are available', async function () {
+                await allSymbolsAreAvailableInOutlineSideBar(JAVA_av_symbols);
+            });
+
+            it('goto symbols', async function () {
+                await allSymbolsAreAvailableInOutlineSideBar(JAVA_av_symbols);
```
Comment (pospisilf) [inline src/ui-test/tests/code.navigation.test.ts:125]: wrong test called

File: src/ui-test/tests/code.navigation.test.ts [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+        beforeEach(async function () {
+            await VSBrowser.instance.openResources(path.join(RESOURCES, CODE_NAVIGATION_XML));
+            await waitUntilEditorIsOpened(driver, CODE_NAVIGATION_XML);
+        });
+
+        afterEach(async function () {
+            await closeEditor(CODE_NAVIGATION_XML, false);
+        });
+
+        describe('Quickpick \'Go to Symbol in Editor\' navigation', function () {
```
Comment (djelinek) [inline src/ui-test/tests/code.navigation.test.ts:72]: is this different test or we can unify naming of this and following describe with Java DSL ones?

**Review by djelinek — CHANGES_REQUESTED** [review]

**Pull request merged**